AprilSylph/XKit-Rewritten
# Implement remove buttons on own post tags

### Description


Core implementation of a feature I've wanted for a long time, but that is somewhat niche: the ability to remove tags from a post with one click. This is very nice when tags are automatically added to posts based on their source (i.e. `#Youtube`) but is otherwise not useful very often.

Resolves #549 (which I intentionally let go stale).

Things one should debate:
- If merging this functionality, should it be part of Quick Tags, be its own feature, be... a tweak (which _probably_ makes the most sense scope-wise but completely violates the norms for how complex a tweak is), etc?
- Should every tag on an editable post be removable or just the auto-added ones? What is the list of auto-added ones?

Ultimately my guess is that no one can come up with a justification for implementing this, but hey, I made it.

<img width="655" src="https://github.com/user-attachments/assets/ab7fb457-0b7a-4551-a1b9-ea7fd0272a23" />


### Testing steps


- Create a post (or multiple posts) with a youtube embed, an instagram embed, a spotify embed, a bandcamp embed, and a soundcloud embed, and some random tags. Save the post; Tumblr will give it extra tags.
- Enable the new Quick Tags option. Confirm that the auto-added tags gain x buttons that will remove the tag in question, but other tags do not.
- Click a remove button. Confirm that the tag is removed.
- Use Quick Tags to add back the same tag (with the same capitalization). Confirm that the tag is added and appears with a remove button.
- Confirm that posts with a relevant tag which the user cannot edit don't have remove buttons.
